fitbenchmarking/fitting/controllers/gsl_controller.py

```diff
@@ -0,0 +1,188 @@
+"""
+Implements a controller for GSL
+https://www.gnu.org/software/gsl/
+using the pyGSL python interface
+https://sourceforge.net/projects/pygsl/
+"""
+
+import pygsl
+from pygsl import multifit_nlin, multiminimize, errno
+from pygsl import _numobj as numx
+
+import numpy as np
+from scipy.optimize._numdiff import approx_derivative
+
+from fitbenchmarking.fitting.controllers.base_controller import Controller
+
+class GSLController(Controller):
+    """
+    Controller for the GSL fitting software
+    """
+
+    def __init__(self, problem, use_errors):
+        """
+        Initializes variable used for temporary storage
+        """
+        super(GSLController, self).__init__(problem, use_errors)
+        self._pinit = None
+        self._solver = None
+
+    def _prediction_error(self, p, data=None):
+        f = self.problem.eval_f(x=self.data_x,
+                                params=p,
+                                function_id=self.function_id)
+        f = f - self.data_y
+        if self.use_errors:
+            f = f / self.data_e
+            
+        return f
+
+    def _jac(self, p, data=None):
+        j = approx_derivative(self._prediction_error,
+                              p)
+        return j
+
+    def _fdf(self, p, data=None):
+        f  = self._prediction_error(p)
+        df = self._jac(p)
+        return f, df
+
+    def _chi_squared(self, p, data=None):
+        f = self._prediction_error(p)
+        return np.dot(f,f)
```

Comment:
My understanding is that the weighted chi squared is (r.r)/omega. This is (r/omega.r/omega)=(r.r)/omega^2.
Please correct me if I've got that wrong

Reply:
Also getting a linting complaint "missing whitespace after ','"

Reply:
I believe what's in the code here is correct (although let's talk through it later to be sure).  This does make me realise that the docs never state precisely what we're testing, so I'll open an issue to address that.